From a91bffb95b75d3c9b0dc0fae4bccc1fd6ec302be Mon Sep 17 00:00:00 2001
From: Rex Lu <rex.lu@mediatek.com>
Date: Wed, 29 Nov 2023 15:51:04 +0800
Subject: [PATCH 2019/2028] mtk: wifi: mt76: mt7996: Refactor rro del ba
 command format

1. remove unused struct
2. refactor upstream del ba command format

Signed-off-by: Rex Lu <rex.lu@mediatek.com>
---
 mt7996/mcu.h | 50 +++-----------------------------------------------
 1 file changed, 3 insertions(+), 47 deletions(-)

diff --git a/mt7996/mcu.h b/mt7996/mcu.h
index 7b8540f6..a05dd6a5 100644
--- a/mt7996/mcu.h
+++ b/mt7996/mcu.h
@@ -273,7 +273,9 @@ struct mt7996_mcu_wed_rro_ba_delete_event {
 	__le16 len;
 
 	__le16 session_id;
-	u8 __rsv2[2];
+	__le16 mld_id;
+	u8 tid;
+	u8 __rsv[3];
 } __packed;
 
 enum  {
@@ -298,52 +300,6 @@ struct mt7996_mcu_thermal_notify {
 	u8 __rsv2[4];
 } __packed;
 
-struct mt7996_mcu_rro_event {
-	struct mt7996_mcu_rxd rxd;
-
-	u8 __rsv1[4];
-
-	__le16 tag;
-	__le16 len;
-} __packed;
-
-struct mt7996_mcu_rro_ba {
-	__le16 tag;
-	__le16 len;
-
-	__le16 wlan_id;
-	u8 tid;
-	u8 __rsv1;
-	__le32 status;
-	__le16 session_id;
-	u8 __rsv2[2];
-} __packed;
-
-struct mt7996_mcu_rro_ba_del_chk_done {
-	__le16 tag;
-	__le16 len;
-
-	__le16 session_id;
-	__le16 mld_id;
-	u8 tid;
-	u8 __rsv[3];
-} __packed;
-
-enum  {
-	UNI_RRO_BA_SESSION_STATUS = 0,
-	UNI_RRO_BA_SESSION_TBL	= 1,
-	UNI_RRO_BA_SESSION_DEL_CHK_DONE = 2,
-	UNI_RRO_BA_SESSION_MAX_NUM
-};
-
-struct mt7996_mcu_rro_del_ba {
-	struct mt7996_mcu_rro_event event;
-
-	u8  wlan_idx;
-	u8  tid;
-	u8 __rsv2[2];
-};
-
 enum mt7996_chan_mib_offs {
 	UNI_MIB_OBSS_AIRTIME = 26,
 	UNI_MIB_NON_WIFI_TIME = 27,
-- 
2.18.0

